test/unittest/dds/publisher/DataWriterTests.cpp

```diff
@@ -450,7 +456,7 @@ TEST(DataWriterTests, ForcedDataSharing)
 
     // DataSharing automatic, unbounded topic data type
     qos = DATAWRITER_QOS_DEFAULT;
-    qos.endpoint().history_memory_policy = fastrtps::rtps::PREALLOCATED_MEMORY_MODE;
```

Comment:
These tests are no testing the default behavior. I think the History Memory Policy should be kept whenever the default behavior is not being tested. Were these tests failing by the behavior change? This comment applies to the other modified tests in this file.

Reply:
I tested reverting all the changes in this file (even those with std::function), and it was still complaining about bad_function_call(). I will keep the lambdas and revert memory policies to its former value of PREALLOCATED_MEMORY_MODE